### PR TITLE
CC-1576/preauth

### PR DESCRIPTION
**Pull request by Ryouzanpaku**
- [x] preauth transaction
- [x] fetch preauth
- [x] webhooks

Basically just added new transaction type that works with the existing authorize method of Unzer class, since it inherits from Authorization class.

### CODE REVIEW

File: src/Resources/TransactionTypes/PreAuthorization.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace UnzerSDK\Resources\TransactionTypes;
+
+use UnzerSDK\Adapter\HttpAdapterInterface;
+
+/**
+ * This represents the authorization transaction.
```
Comment (m-xhixha) [inline src/Resources/TransactionTypes/PreAuthorization.php:8]: Did you mean preAuthorization here ?

File: src/Services/PaymentService.php [hunk 2]
```diff
@@ -104,6 +104,38 @@ public function updateAuthorization($payment, Authorization $authorization): Aut
...
 
+    public function performPreAuthorization(
+        Authorization $authorization,
+                      $paymentType,
+                      $customer = null,
+        Metadata      $metadata = null,
+        Basket        $basket = null
+    ): Authorization
+    {
+        $payment = $this->createPayment($paymentType);
```
Comment (m-xhixha) [inline src/Services/PaymentService.php:115]: In general, what exactly do we do here ?

Comment (Ryouzanpaku) [reply src/Services/PaymentService.php:115]: We are preparing payment object that hold and connect all relevant payment data for the payment request. This also makes sure that a viable type object or id is used.
But i just realized that this method is not actually used so i will remove it. Reason is that the existing authorization method works just fine with preauth type.

Comment (Ryouzanpaku) [reply src/Services/PaymentService.php]: removed method. reusing existing one

File: src/Resources/TransactionTypes/PreAuthorization.php (hunk 1 above)
Comment (Ryouzanpaku) [reply src/Resources/TransactionTypes/PreAuthorization.php]: yes, its updated now.

**Review by m-xhixha — APPROVED** [review]

**Pull request merged**